From d2f2727ea24c7c3127e510d930ceb3019ed86bb9 Mon Sep 17 00:00:00 2001
From: Cheick Traore <cheick.traore-ext@st.com>
Date: Wed, 14 Aug 2024 10:15:04 +0200
Subject: [PATCH 1096/1695] dts: stm32: Change LPUART clock on stm32mp257f-dk
 board

Change LPUART clock from PLL5 to MSI_KER to use the wakeup feature of the
LPUART. MSI_KER is the only one clock ables to be used in low-power by
LPUART.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e2fb9231a364f4b34c0eb76f7aed3dab0656906a dts: stm32: Change LPUART clock on stm32mp257f-dk board]
Change-Id: I7f15903a65b5b60a9651fb6bd8f3ea0017122a0a
Signed-off-by: Cheick Traore <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/398395
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
index 9529f3c2d..bd4edcc7c 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
@@ -73,7 +73,7 @@
 		FLEXGEN_CFG(36, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(37, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(38, XBAR_SRC_PLL5, 0, 3)
-		FLEXGEN_CFG(39, XBAR_SRC_PLL5, 0, 3)
+		FLEXGEN_CFG(39, XBAR_SRC_MSI_KER, 0, 0)
 		FLEXGEN_CFG(40, XBAR_SRC_LSE, 0, 0)
 		FLEXGEN_CFG(41, XBAR_SRC_PLL4, 0, 11)
 		FLEXGEN_CFG(42, XBAR_SRC_PLL7, 0, 16)
-- 
2.47.2

